hornetq-server/src/main/java/org/hornetq/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -1048,7 +1049,8 @@ public void processRoute(final ServerMessage message, final RoutingContext conte
 
             // We need to kick delivery so the Queues may check for the cursors case they are empty
             schedulePageDelivery(tx, entry);
-            continue;
+
+            paged = true;
```

Comment:
Please revert this change... also remove the setPaged on PagedReference, and MessageReference... a PagedReference will be an instance of PagedRefrenceImpl.


you may add isPaged if it doesn't exist yet. and always return false on MessageReferenceIMpl and PagedReferenceImpl.